File: Arrays/pivot_index.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
            left_sum = 0
            right_sum = 0 
            for num in nums:
                right_sum += num
            for i in range(len(nums)):
                left_sum += nums[i] 
                if left_sum == right_sum:
                    return i
                right_sum -= nums[i] 
            return -1

File: Arrays/pivot_index.ts
function pivotIndex(nums: number[]): number {
    let leftSum: number = 0;
    let rightSum: number = 0;
    for(let num of nums){
        rightSum += num
    }
    for(let i = 0; i < nums.length; i++){
        leftSum += nums[i];
        if(leftSum === rightSum) return i
        rightSum -= nums[i]
    }
    return -1
};
